packages/alchemy/src/config/actions/watchAccount.ts

```diff
@@ -0,0 +1,30 @@
+import { ClientOnlyPropertyError } from "../errors.js";
+import type { AlchemyAccountsConfig, SupportedAccountTypes } from "../types";
+import type { GetAccountResult } from "./getAccount";
+
+export const watchAccount =
+  <TAccount extends SupportedAccountTypes>(
+    type: TAccount,
+    config: AlchemyAccountsConfig
+  ) =>
+  (onChange: (account: GetAccountResult<TAccount>) => void) => {
+    if (config.clientStore == null) {
+      throw new ClientOnlyPropertyError("account");
+    }
+
+    return config.clientStore.subscribe(
+      ({ accounts }) => accounts[type],
+      onChange,
+      {
+        fireImmediately: true,
+        equalityFn(a, b) {
+          return (
+            a?.status === b?.status ||
+            (a?.status === "READY" &&
+              b?.status === "READY" &&
+              a.account.address === b.account.address)
```

Comment:
oops yea this is wrong.

The comparison is meant to be for status === ready also compare their addresses otherwise only compare their statuses... which means I need to swap the order here